core/src/orderbook/streamed/block_timestamp_reading.rs

```diff
@@ -117,12 +117,17 @@ impl<T: BlockTimestampBatchReading + Send> CachedBlockTimestampReader<T> {
 }
 
 impl<T: BlockTimestampReading + Send> BlockTimestampReading for CachedBlockTimestampReader<T> {
-    fn block_timestamp(&mut self, block_hash: H256) -> BoxFuture<Result<u64>> {
+    fn block_timestamp(&mut self, block: BlockId) -> BoxFuture<Result<u64>> {
         async move {
+            let block_hash = match block {
+                BlockId::Hash(hash) => hash,
+                _ => return self.inner.block_timestamp(block).await,
```

Comment:
I assume we don't want to cache the timestamps of block numbers, since they can change? Maybe we can later implement a cache if the timestamp is sufficiently long in the pas to be considered "final". I'm a little bit worried that consumers of a `CachedBlockTimestampReader` would expect the implementation to cache all types of blockIds.

Reply:
The issue with caching by block number is that in reorgs the cache may become invalidated.

Reply:
I think we can fix this if we are sensitive to reorgs by having a number of confirmation blocks before committing the timestamp to the cache.